client/lib/wp/sync-handler/index.js

```diff
@@ -161,10 +142,48 @@ export class SyncHandler {
 			};
 
 			/**
+			 * clear pageSeries if pagination out of alignment
+			 *
+			 * @param {Object} combinedResponse - object with local and server responses
+			 * @returns {Object} - combinedResponse object
+			 */
+			const adjustPagination = combinedResponse => {
```

Comment:
Just a note. We will have to be completely aware of how these Promises are joined and how we are handling their function parameters.

```es6
this.retrieveRecord( key ) // -> localforage record
  then( localResponseHandler, recordErrorHandler ) // -> localforage record
  .then( networkFetch ) // -> object with two localforage records { localResponse, serverResponse }
  .then( adjustPagination, networkErrorHandler ) // -> same above { localResponse, serverResponse }
  .then( cacheResponse );
```